components/camel-oauth/src/main/java/org/apache/camel/oauth/InMemorySessionStore.java

```diff
@@ -93,4 +96,15 @@ private Map<String, String> getCookies(Exchange exchange) {
 
         return cookieMap;
     }
+
+    private void setSessionCookie(Message msg, OAuthSession session) {
+        var sessionId = session.getSessionId();
+        var cookieId = "%s=%s".formatted(CAMEL_OAUTH_COOKIE, sessionId);
+        if (msg.getHeader("Set-Cookie") != null) {
+            throw new IllegalStateException("Duplicate 'Set-Cookie' header");
+        }
+        var cookie = cookieId + "; Path=/; HttpOnly; SameSite=None; Secure";
+        msg.setHeader("Set-Cookie", cookie);
+        log.info("Set-Cookie: {}", cookie);
```

Comment:
The INFO logging may need to be toned down to DEBUG as for production usage you may not want to log all interactions